futures-util/src/async_await/join.rs

```diff
@@ -28,23 +28,21 @@ macro_rules! join {
             let mut $fut = $crate::future::maybe_done($fut);
             $crate::pin_mut!($fut);
         )*
-        loop {
+        await!($crate::future::poll_fn(move |cx| {
             let mut all_done = true;
             $(
-                if $crate::poll!($fut.reborrow()).is_pending() {
+                if $crate::future::Future::poll($fut.reborrow(), cx).is_pending() {
```

Comment:
It's safe to use the method call syntax because we know that it's a `MaybeDone` future.

Reply:
I know this. It's fine here because this type is under our control because we define it.

Reply:
@MajorBreakfast the problem is the user _could_ introduce a new trait that has a `poll` method and implement that for `MaybeDone`. That’s highly unlikely, but the only way to guarantee no method resolution errors is to always use expanded function calls.

Reply:
Ah, I see. And it's okay for `reborrow` because it's an inherent method and those have precedence (I just [checked](http://play.rust-lang.org/?gist=c0e79343117a383e92092046f6e5ad9b&version=stable&mode=debug&edition=2015) that because I was curious)

Reply:
Yup, as others said we can't assume that the `Future` trait is in-scope or that other methods called `poll` haven't been provided via other traits. `.reborrow()` is fine because it's an inherent method.